azext_edge/tests/edge/orchestration/resources/connector/opcua/test_opcua_certs_unit.py

```diff
@@ -0,0 +1,913 @@
+            method=responses.PUT,
+            url=get_secret_endpoint(keyvault_name="mock-keyvault", secret_name="certificate-pem"),
+            json={},
+            status=200,
+            content_type="application/json",
+        )
+
+        # set opcua secretsync
+        mocked_responses.add(
+            method=responses.PUT,
+            url=get_secretsync_endpoint(
+                secretsync_name=OPCUA_CLIENT_CERT_SECRET_SYNC_NAME, resource_group_name=rg_name
+            ),
+            json=expected_secret_sync,
+            status=200,
+            content_type="application/json",
+        )
+
+    result = None
+
+    try:
+        result = add_connector_opcua_client(
+            cmd=mocked_cmd,
+            instance_name=instance_name,
+            resource_group=rg_name,
+            public_key_file=public_file_name,
+            private_key_file=private_file_name,
+            application_uri="uri",
+            subject_name="subjectname",
+        )
+    except Exception as e:
+        private_file_name = os.path.basename(private_file_name).replace(".pem", "")
+        public_file_name = os.path.basename(public_file_name).replace(".der", "")
+        if private_file_name != public_file_name:
+            assert "Public key file pubkey and private key file prikey must match" in str(e)
+            return
+
+    if not expected_resources_map["custom locations"]:
+        assert (
+            mocked_logger.error.call_args[0][0] == f"Secret sync is not enabled for the instance {instance_name}. "
+            "Please enable secret sync before adding certificate."
+        )
+        return
+
+    if not expected_resources_map["extension"]:
+        assert mocked_logger.error.call_args[0][0] == "IoT Operations extension not found."
+        return
+
+    if result:
+        if not client_app_spc:
+            assert (
+                mocked_logger.warning.call_args[0][0] == f"Azure Key Vault Secret Provider Class {OPCUA_SPC_NAME} "
+                "not found, creating new one..."
+            )
+
+        if not client_app_secretsync:
+            assert (
+                mocked_logger.warning.call_args[0][0] == f"Secret Sync {OPCUA_CLIENT_CERT_SECRET_SYNC_NAME} "
+                "not found, creating new one..."
+            )
+        mocked_resource_map().connected_cluster.get_extensions_by_type.assert_called_once_with(
+            "microsoft.iotoperations"
+        )
+        mocked_resource_map().connected_cluster.update_aio_extension.assert_called_once_with(
+            extension_name=expected_resources_map["extension"][IOT_OPS_EXTENSION_TYPE]["name"],
+            properties={
+                "configurationSettings": {
+                    "connectors.values.securityPki.applicationCert": OPCUA_CLIENT_CERT_SECRET_SYNC_NAME,
+                    "connectors.values.securityPki.subjectName": "subjectname",
+                    "connectors.values.securityPki.applicationUri": "uri",
+                }
+            },
```

Comment:
very long file that will prob become longer with more CRUD commands - you may want to break it up into 3 files